src/main/java/org/jenkinsci/plugins/build_token_root/BuildRootAction.java

```diff
@@ -150,4 +167,57 @@ private void ok(StaplerResponse rsp) throws IOException {
         w.close();
     }
 
+    // --- adapter methods for Job type which do follow AbstractProject semantic but don't inherit AbstractProject.
+
+    private BuildAuthorizationToken getAuthToken(Job<?, ?> p) {
+        if (p instanceof AbstractProject) return ((AbstractProject) p).getAuthToken();
+
+        try {
+            Method m = p.getClass().getMethod("getAuthToken");
+            return (BuildAuthorizationToken) m.invoke(p);
+        } catch (Exception e) {
+            // Job does not support BuildAuthorizationToken
+        }
+        return null;
+    }
+
+    private boolean isDisabled(Job<?, ?> p) {
+        if (p instanceof AbstractProject) return ((AbstractProject) p).isDisabled();
+
+        try {
+            Method m = p.getClass().getMethod("isDisabled");
+            return (Boolean) m.invoke(p);
+        } catch (Exception e) {
+            // Job can't be disabled
+        }
+        return false;
+    }
+
+    private int getDelay(Job<?, ?> p, StaplerRequest req) throws ServletException {
+        if (p instanceof AbstractProject) return ((AbstractProject) p).getDelay(req);
+
+        try {
+            Method m = p.getClass().getMethod("getDelay", int.class);
```

Comment:
There is no such method, so this will always throw an exception. I guess you meant `getDelay(StaplerRequest)`, but anyway this is deprecated. Use `TimeDuration` instead.